src/Symfony/Component/Routing/Route.php

```diff
@@ -553,12 +553,18 @@ private function extractInlineDefaultsAndRequirements(string $pattern): string
 
     private function sanitizeRequirement(string $key, string $regex)
     {
-        if ('' !== $regex && '^' === $regex[0]) {
-            $regex = (string) substr($regex, 1); // returns false for a single character
```

Comment:
the string cast is indeed not necessary anymore since php 7.0. before that substr returned false for `substr('^', 1)`

Reply:
But Symfony requires php >=7.2.5.